### DISCUSSION TITLE
Flags which override required options?

### DISCUSSION THREAD

**Discussion by trapexit** [CLIUtils/CLI11#945] [Q&A] [answered]
It is common to have a flag like `-v,--version` which causes the app to print the version and exit. My app has an otherwise required positional argument. Is there any way to make the parser ignore the requirement only if the flag is set or do I need to handle it manually? Using a group makes the auto generated error odd and would be confusing for most users.

**Comment by phlptp** [comment, marked as answer]
I won't say it is common.  "version" and "help" are somewhat common which is why they have special handlers to them.  If you want to do something like that you can follow that structure with an exception that gets thrown and you handle it manually in your code.   

option_groups were meant to handle things like that but if you don't like the error output or help printing for that, the other option would be a custom callback for the flag and setting the trigger_on_parse flag so the callback is executed immediately.  That callback can change the `required` modifier on the other option.
